src/signature/mod.rs

```diff
@@ -24,6 +30,45 @@ impl TryFrom<u8> for SignatureVersion {
     }
 }
 
+fn create_ed25519_keypair_from_keys(
+    pubkey: &[u8],
+    seckey: &[u8],
+) -> Result<Ed25519Keypair, SigningError> {
+    let pair_bytes: Vec<u8> = [seckey.to_vec(), pubkey.to_vec()].concat();
+
+    Ed25519Keypair::from_bytes(&pair_bytes).map_err(|_| SigningError::InvalidEd25519Key)
+}
+
+/// Sign an ed25519 signature
+pub fn sign_ed25519(
+    public_key: &[u8],
+    secret_key: &[u8],
+    payload: &[u8],
+) -> Result<Vec<u8>, SigningError> {
+    create_ed25519_keypair_from_keys(public_key, secret_key)
+        .map(|keypair| keypair.sign(payload).to_bytes().to_vec())
+}
+
+/// Sign an sr25519 signature
+pub fn sign_sr25519(
+    public_key: &[u8],
+    secret_key: &[u8],
+    payload: &[u8],
+) -> Result<Vec<u8>, SigningError> {
+    Sr25519SecretKey::from_ed25519_bytes(secret_key)
```

Comment:
Need to check the pub key format also instead of `unwrap` incase it fails.
This style may help
```rust

let secret_key_ = Sr25519SecretKey::from_ed25519_bytes(secret_key).map_err(|_| InvalidSr25519SecretKey)?;
let public_key_ = Sr25519PublicKey::from_bytes(public_key).map_err(InvalidSr25519PublicKey)?;

secret_key_.sign_simple(CONTEXT_ID, payload, &public_key_)
  .to_bytes()
  .to_vec()
```

Reply:
@holygits In Rust, style of `let xxxx_ = xxx;` is our suggested naming convention for local variable? How about private functions, like `fn _xx(() {...}`?

Reply:
It's not convention couldn't think of a better name 🤷‍♀️